src/general/neighborhood_search.jl

```diff
@@ -15,8 +15,8 @@ function PointNeighbors.foreach_point_neighbor(f, system::GPUSystem, neighbor_sy
                                                neighborhood_search;
                                                points=each_moving_particle(system),
                                                parallel=true)
-    @threaded system for point in points
-        PointNeighbors.foreach_neighbor(f, system_coords, neighbor_coords,
-                                        neighborhood_search, point)
-    end
+    # For `GPUSystem`s, explicitly pass the backend, so that with a `GPUSystem` with CPU
+    # backend, it will actually launch the KernelAbstractions.jl kernels on the CPU.
```

Comment:
```suggestion
# For `GPUSystem`s, explicitly pass the backend, so that a `GPUSystem` with a CPU
# backend will actually launch the KernelAbstractions.jl kernels on the CPU.
```